addon/bootstrap.js

```diff
@@ -0,0 +1,118 @@
+/* globals AddonManager, Components, Services */
+
+const OLD_ADDON_PREF_NAME = "extensions.jid1-NeEaf3sAHdKHPA@jetpack.deviceIdInfo";
+const OLD_ADDON_ID = "jid1-NeEaf3sAHdKHPA@jetpack";
+const ADDON_ID = "screenshots@mozilla.org";
+const TELEMETRY_PREF = "toolkit.telemetry.enabled";
+const PREF_BRANCH = "extensions.screenshots.";
+const USER_DISABLE_PREF = "extensions.screenshots.disabled";
+const SYSTEM_DISABLE_PREF = "extensions.screenshots.system-disabled";
+
+const { interfaces: Ci, utils: Cu } = Components;
+Cu.import("resource://gre/modules/AddonManager.jsm");
+Cu.import("resource://gre/modules/Console.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+const { EmbeddedExtensionManager } = Cu.import("resource://gre/modules/LegacyExtensionsUtils.jsm");
+
+const prefs = Services.prefs;
+const prefObserver = {
+  register: function() {
+    prefs.addObserver(PREF_BRANCH, this, false);
+  },
+
+  unregister: function() {
+    prefs.removeObserver(PREF_BRANCH, this);
+  },
+
+  observe: function(aSubject, aTopic, aData) {
+    // aSubject is the nsIPrefBranch we're observing (after appropriate QI)
+    // aData is the name of the pref that's been changed (relative to aSubject)
+    if (aData == USER_DISABLE_PREF || aData == SYSTEM_DISABLE_PREF) {
+      handleStartup();
+    }
+  }
+};
+
+function startup(data, reason) { // eslint-disable-line no-unused-vars
+  prefObserver.register();
+  handleStartup();
+}
+
+function shutdown(data, reason) { // eslint-disable-line no-unused-vars
+  prefObserver.unregister();
+}
+
+function install(data, reason) {} // eslint-disable-line no-unused-vars
+
+function uninstall(data, reason) {} // eslint-disable-line no-unused-vars
+
+function getBoolPref(pref) {
+  return prefs.getPrefType(pref) && prefs.getBoolPref(pref);
```

Comment:
You can now pass a second argument to getBoolPref to specify a default if the pref doesn't exist.